README.md

```diff
@@ -69,36 +69,38 @@ by all `valuePath` in `columns` array.
 The following template renders a simple table.
 
 ```
-  {{#ember-table
-    rows=rows
-    columns=columns
-    as |row|
-  }}
-    {{#ember-table-row
-      row=row
-      as |cell|
-    }}
-      {{cell.value}}
-    {{/ember-table-row}}
+  {{#ember-table as |t|}}
+    {{ember-thead api=t columns=columns}}
+
+    {{ember-tbody api=t rows=rows}}
   {{/ember-table}}
 ```
 
-To have your own custom cell, pass in the cell component name in each element of `columns` array and
-specify it in the template.
+You can use the block form of the table to customize its template. The component
+structure matches that of actual HTML tables, and allows you to customize it at
+any level. At the cell level, you get access to these four values:
+
+* `value` - The value of the cell
+* `cell` - A unique cell cache. You can use this to track cell state without
+  dirtying the underlying model.
+* `column` - The column itself.
+* `row` - The row itself.
+
+You can use these values to customize cell in many ways. For instance, if you
+want to have every cell in a particular column use a component, you can add a
+`component` field to your column (or feel free to use any other property name
+you like):
 
 ```
-  {{#ember-table
-    rows=rows
-    columns=columns
-    as |row|
-  }}
-    {{#ember-table-row
-      row=row
-      as |cell|
-    }}
-      {{#component cell.column.cellComponent
-        cell=cell
-      }}
+  {{#ember-table as |t|}}
+    {{ember-thead api=t columns=columns}}
```

Comment:
maybe `table=t` instead of `api=t`?

Reply:
Yeah I've been going back and forth on that one. On the one hand, it reads better for now, but on the other I think it may actually obfuscate what we're passing in and why. This especially becomes an issue later on:

```
{{#ember-tbody as |b|}}
  {{#ember-tr body=b as |r|}}
    {{#ember-td row=r as |value cell column row|}}
       ^r and row are not the same thing
    {{/ember-td}}
  {{/ember-tr}}
{{/ember-tbody}}
```

I think it may make sense to add the name as an alias to each component, so users can choose whichever one makes more sense to them and we can see which is less confusing over time. Once we get to 2.3 we'll be converting to contextual components anyways, so it's not really a big deal either way.